MO4/Sniffs/Formatting/AlphabeticalUseStatementsSniff.php

```diff
@@ -280,12 +293,82 @@ private function findNewDestination(
             $prevLine      = $tokens[$prevPtr]['line'];
             $prevImportArr = $this->getUseImport($phpcsFile, $prevPtr);
         } while ($prevLine === ($line - 1)
-            && (strcmp($prevImportArr['content'], $import) > 0)
+            && ($this->compareString($prevImportArr['content'], $import) > 0)
         );
 
         return $ptr;
 
     }//end findNewDestination()
 
 
+    /**
+     * Compare namespace strings according defined order function.
+     *
+     * @param string $a first namespace string
+     * @param string $b second namespace string
+     *
+     * @return int
+     */
+    private function compareString($a, $b)
+    {
+        if ('dictionary' === $this->order) {
+            return $this->dictionaryCompare($a, $b);
+        } else if ('string' === $this->order) {
+            return strcmp($a, $b);
+        } else if ('string-locale' === $this->order) {
+            return strcoll($a, $b);
+        } else if ('string-case-insensitive' === $this->order) {
+            return strcasecmp($a, $b);
+        } else {
+            return $this->dictionaryCompare($a, $b);
```

Comment:
This method falls back to `dictionary` as validation approach by default. I've seen that PHPCS loves to mutate public properties using an XML, however I wonder if some kind of validation (e.g. throw an exception in here if `order` has an invalid value) makes sense to avoid unexpected behavior

Reply:
Added the check to `register()`